apis/datadoghq/v2alpha1/test/builder.go

```diff
@@ -157,13 +159,85 @@ func (builder *DatadogAgentBuilder) initAdmissionController() {
 		builder.datadogAgent.Spec.Features.AdmissionController = &v2alpha1.AdmissionControllerFeatureConfig{}
 	}
 }
+func (builder *DatadogAgentBuilder) initSidecarInjection() {
+	if builder.datadogAgent.Spec.Features.AdmissionController.AgentSidecarInjection == nil {
+		builder.datadogAgent.Spec.Features.AdmissionController.AgentSidecarInjection = &v2alpha1.AgentSidecarInjectionConfig{}
+	}
+}
 
 func (builder *DatadogAgentBuilder) WithAdmissionControllerEnabled(enabled bool) *DatadogAgentBuilder {
 	builder.initAdmissionController()
 	builder.datadogAgent.Spec.Features.AdmissionController.Enabled = apiutils.NewBoolPointer(enabled)
 	return builder
 }
 
+func (builder *DatadogAgentBuilder) WithMutateUnlabelled(enabled bool) *DatadogAgentBuilder {
+	builder.initAdmissionController()
+	builder.datadogAgent.Spec.Features.AdmissionController.MutateUnlabelled = apiutils.NewBoolPointer(enabled)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithAdmissionControllerRegistry(registryName string) *DatadogAgentBuilder {
+	builder.initAdmissionController()
+	builder.datadogAgent.Spec.Features.AdmissionController.Registry = apiutils.NewStringPointer(registryName)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithAgentCommunicationMode(options string) *DatadogAgentBuilder {
+	builder.initAdmissionController()
+	builder.datadogAgent.Spec.Features.AdmissionController.AgentCommunicationMode = apiutils.NewStringPointer(options)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithServiceName(name string) *DatadogAgentBuilder {
+	builder.initAdmissionController()
+	builder.datadogAgent.Spec.Features.AdmissionController.ServiceName = apiutils.NewStringPointer(name)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithWebhookName(name string) *DatadogAgentBuilder {
+	builder.initAdmissionController()
+	builder.datadogAgent.Spec.Features.AdmissionController.WebhookName = apiutils.NewStringPointer(name)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithAdmissionControllerFailurePolicy(failurePolicyName string) *DatadogAgentBuilder {
+	builder.initAdmissionController()
+	builder.datadogAgent.Spec.Features.AdmissionController.FailurePolicy = apiutils.NewStringPointer(failurePolicyName)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithSidecarInjectionSetup(sidecarConfig v2alpha1.AgentSidecarInjectionConfig) *DatadogAgentBuilder {
```

Comment:
I doubt this is the intention of the builder. Idea is to make creation of DDA with mix of configs easy by having granular functions. This one takes an instance of DDA type, something it should create